packages/slides-frontend/src/containers/redux-store/DeckReducer/reducer.ts

```diff
@@ -47,6 +84,37 @@ const DeckState = (state: Deck = initialDeck, action: Action): Deck =>
         } else alert('Not possible to remove the only slide');
         break;
       }
+      case CLONE_SLIDE: {
+        // create a new blank slide
+        let slide:Slide = {
+          ID: uuidv4(),
+          itemsArray: [],
+        }
+        // with a loop over the current itemsArray, create and push newItems in the new itemsArray
+        slide.itemsArray = draft.slides[draft.currentSlide].itemsArray.map(currentItem => {
+          const copyObj = JSON.parse(JSON.stringify(currentItem));
```

Comment:
if currentItem is an object you could do something like `const copyObj = {...currentItem}` to copy an object. But keep in mind this will perform a shallow copy

Reply:
I need a deep copy, that's why I do it like that.
The `currentItem` object has Size and Position sub-objects that will not be identified as objects if I do it another way.
They will remain `Proxies`

Reply:
Yeah, I am aware of it. It is just that it looks a bit ugly IMO :P You could write a custom function (which wouldn't be hard to do) that would perform a deep clone on an object.

Reply:
Ok fixed it